extensions/amp-consent/0.1/consent-policy-manager.js

```diff
@@ -14,39 +14,177 @@
  * limitations under the License.
  */
 
+import {CONSENT_ITEM_STATE} from './consent-state-manager';
+import {dev} from '../../../src/log';
+import {getServicePromiseForDoc} from '../../../src/service';
+
+const CONSENT_STATE_MANAGER = 'consentStateManager';
+const TAG = 'consent-policy-manager';
+
 /**
  * Possible consent policy state to proceed with.
  * @enum {number}
  */
 export const CONSENT_POLICY_STATE = {
-  SUFFIENT: 0,
+  SUFFICIENT: 0,
   INSUFFICIENT: 1,
   UNKNOWN: 2,
 };
 
 export class ConsentPolicyManager {
   constructor(ampdoc) {
+    /** @private {!../../../src/service/ampdoc-impl.AmpDoc} */
     this.ampdoc_ = ampdoc;
 
-    this.defaultPolicy_ = null;
+    /** @private {!Object<string, ?Promise>} */
+    this.policyInstancePromise_ = {};
+
+    /** @private {!Object<string, ?function()>} */
+    this.policyInstancePromiseResolver_ = {};
+
+    /** @private {!Object<string, ConsentPolicyInstance>} */
+    this.instances_ = {};
 
-    this.policyInstances_ = {};
+    /** @private {!Promise} */
+    this.ConsentStateManagerPromise_ =
+        getServicePromiseForDoc(this.ampdoc_, CONSENT_STATE_MANAGER);
   }
 
   /**
    * Register the policy instance
-   * @param {string} unusedPolicyId
+   * Example policy config format:
+   * {
+   *   "waitFor": {
+   *     "consentABC": undefined,
+   *     "consentDEF": undefined
+   *   }
+   * }
+   *
+   * TODO: Add support to timeout
+   * @param {string} policyId
+   * @param {!JsonObject} config
+   */
+  registerConsentPolicyInstance(policyId, config) {
+    dev().assert(!this.instances_[policyId],
+        `${TAG}: instance already registered`);
+
+    const waitFor = Object.keys(config['waitFor'] || {});
```

Comment:
Why isn't this an array to begin with?

Reply:
We make it an object because it is supposed to be something like
```
{
  'ABC': ['A', 'B'],
  'DEF': ['D']
}
```
in future.